packages/hash/src/index.js

```diff
@@ -1,47 +1,57 @@
 // @flow
 /* eslint-disable */
-// murmurhash2 via https://github.com/garycourt/murmurhash-js/blob/master/murmurhash2_gc.js
+// Inspired by https://github.com/levitation/murmurhash-js
+// Ported from https://github.com/aappleby/smhasher/blob/61a0530f28277f2e850bfc39600ce61d02b518de/src/MurmurHash2.cpp#L37-L86
 
-export default function murmurhash2_32_gc(str: string) {
-  var l = str.length,
-    h = l ^ l,
-    i = 0,
-    k
+export default function murmur2(str: string) {
+  // 'm' and 'r' are mixing constants generated offline.
+  // They're not really 'magic', they just happen to work well.
 
-  while (l >= 4) {
+  // const m = 0x5bd1e995;
+  // const r = 24;
+
+  // Initialize the hash to a 'random' value
+
+  var len = str.length,
+    h = len ^ len;
```

Comment:
This is always equal to 0 (`len ^ len === 0`). Can we just use 0 here? "Initialize the hash to a 'random' value" seems to be incorrect here - original algo initializes this using a received `seed` value (making it random~), but we dont have this here.